### ISSUE TITLE
Feature Request: GraphQL API request need copyWith method

### ISSUE THREAD

**Issue by MarlonJD** [aws-amplify/amplify-flutter#4364]
### Description

Hello, in GraphQL API, we need custom document, custom variables, authModes etc. for searching or getting more information in complex schema, whenever we are doing this in modification, we have to declare all GraphQLRequest again, so easily what if we create copyWith method for this?

example modified request:
```
final request = ModelQueries.get<Expense>(
      Expense.classType,
      ExpenseModelIdentifier(id: id!),
    );

    const modifiedDocument = r"""query getExpense($id: ID!) {
  getExpense(id: $id) {
    id
    total
    tax
   ...
}""";

    final modifiedRequest = GraphQLRequest<Expense>(
      document: modifiedDocument,
      apiName: request.apiName,
      authorizationMode: request.authorizationMode,
      variables: request.variables,
      headers: request.headers,
      decodePath: request.decodePath,
      modelType: request.modelType,
    );

  final response = await Amplify.API
        .query(
          request: modifiedRequest,
        )
        .response;
```

but if we had `copyWith` for request we could just write this:

```
final request = ModelQueries.get<Expense>(
      Expense.classType,
      ExpenseModelIdentifier(id: id!),
    );

    const modifiedDocument = r"""query getExpense($id: ID!) {
  getExpense(id: $id) {
    id
    total
    tax
   ...
}""";

    final modifiedRequest = request.copyWith(
        document: modifiedDocument,
    );

  final response = await Amplify.API
        .query(
          request: modifiedRequest,
        )
        .response;
```

what do you think about this?

### Categories

- [ ] Analytics
- [ ] API (REST)
- [X] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Platforms

- [X] iOS
- [X] Android
- [X] Web
- [X] macOS
- [X] Windows
- [X] Linux

### Flutter Version

3.3.10

### Amplify Flutter Version

0.6.10

### Deployment Method

Amplify CLI

**Comment by Equartey** [comment]
Hi @MarlonJD, 

Thanks for the feature request. 

I agree this is a worthwhile addition. In fact, we've discussed adding something similar before. 

We'll use this feature request to track interest. I encourage others to leave a reaction which will help us gage interest and allow us to prioritize properly.

**Comment by MarlonJD** [comment]
Thanks for your quick response, I just added PR for this, I tested and it works, less code. I hope we can merge this feature soon

**Comment by Equartey** [comment]
Commenting for visibility. This is a duplicate of https://github.com/aws-amplify/amplify-flutter/issues/2927, however will leave this one open because of the pull request.

**Comment by Equartey** [comment]
The PR has been merged and will be included in the next release.

**Comment by Equartey** [comment]
Hi @MarlonJD, closing this as we have released it in [Amplify Flutter 1.7.0](https://pub.dev/packages/amplify_flutter/changelog#170).

**Comment by MarlonJD** [comment]
> Hi @MarlonJD, closing this as we have released it in [Amplify Flutter 1.7.0](https://pub.dev/packages/amplify_flutter/changelog#170).

Hello, I'm happy to see it released, Thanks 🎉
